plugins/pulp_rpm/plugins/distributors/yum/metadata/modules.py

```diff
@@ -0,0 +1,37 @@
+import hashlib
+import os
+
+from pulp.plugins.util.metadata_writer import MetadataFileContext
+from pulp.server.exceptions import PulpCodedException
+
+from pulp_rpm.common.constants import CONFIG_DEFAULT_CHECKSUM
+from pulp_rpm.plugins import error_codes
+from pulp_rpm.plugins.distributors.yum.metadata.metadata import REPO_DATA_DIR_NAME
+from pulp_rpm.yum_plugin import util
+
+
+_LOG = util.getLogger(__name__)
+
+MODULES_FILE_NAME = 'modules.yaml.gz'
+
+
+# Note that this is "MetadataFileContext" from the core plugin API, not from
+# pulp_rpm.plugin.distributors.yum.metadata.metadata
+class ModulesFileContext(MetadataFileContext):
+
+    def __init__(self, working_dir, checksum_type=CONFIG_DEFAULT_CHECKSUM):
+        metadata_file_path = os.path.join(working_dir, REPO_DATA_DIR_NAME, MODULES_FILE_NAME)
+        super(ModulesFileContext, self).__init__(metadata_file_path, checksum_type)
+
+    def initialize(self):
+        super(ModulesFileContext, self).initialize()
+
+    def finalize(self):
+        super(ModulesFileContext, self).finalize()
+
+    def add_document(self, document, doc_checksum):
+        checksum = hashlib.sha256(document).hexdigest()
```

Comment:
@dralley , it seems like that any checksum type except the sha256 triggers exception, doesn't it?

Reply:
I didn't experience that when I tested w/ checksum-type=sha1.  I believe this particular checksum (of the content) is always sha256.  It's for internal validation and isn't used in the name of the file or anything.

https://github.com/pulp/pulp_rpm/blob/2-master/plugins/pulp_rpm/plugins/db/models.py#L1808